--- a/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h	2022-12-14 02:29:58.157700300 +0100
+++ b/net/third_party/quiche/src/quiche/quic/core/crypto/quic_hkdf.h	2023-05-10 12:58:39.019144897 +0200
@@ -6,6 +6,7 @@
 #define QUICHE_QUIC_CORE_CRYPTO_QUIC_HKDF_H_
 
 #include <vector>
+#include <cstdint>
 
 #include "absl/strings/string_view.h"
 #include "quiche/quic/platform/api/quic_export.h"
--- a/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h	2023-05-10 13:25:39.984693660 +0200
+++ b/net/third_party/quiche/src/quiche/quic/core/quic_connection_id.h	2023-05-10 13:25:52.902019356 +0200
@@ -7,6 +7,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "absl/types/span.h"
 #include "quiche/quic/platform/api/quic_export.h"
