description: build using system openjpeg

--- a/third_party/pdfium/core/fpdfapi/render/BUILD.gn
+++ b/third_party/pdfium/core/fpdfapi/render/BUILD.gn
@@ -49,6 +49,7 @@ jumbo_source_set("render") {
   ]
   configs += [ "../../../:pdfium_core_config" ]
   deps = [
+    "../../../third_party:libopenjpeg2",
     "../../../constants",
     "../../fxcodec",
     "../../fxcrt",
--- a/third_party/pdfium/core/fpdfapi/BUILD.gn
+++ b/third_party/pdfium/core/fpdfapi/BUILD.gn
@@ -12,6 +12,7 @@
   ]
   configs += [ "../../:pdfium_core_config" ]
   deps = [
+    "../../third_party:libopenjpeg2",
     "../fxcodec",
     "../fxcrt",
     "../fxge",
--- a/third_party/pdfium/core/fpdfapi/page/BUILD.gn
+++ b/third_party/pdfium/core/fpdfapi/page/BUILD.gn
@@ -97,6 +97,7 @@
   ]
   configs += [ "../../../:pdfium_core_config" ]
   deps = [
+    "../../../third_party:libopenjpeg2",
     "../../../constants",
     "../../fdrm",
     "../../fxcodec",
--- a/third_party/pdfium/pdfium.gni
+++ b/third_party/pdfium/pdfium.gni
@@ -63,7 +63,7 @@
   use_system_lcms2 = false
 
   # Don't build against bundled libopenjpeg2.
-  use_system_libopenjpeg2 = false
+  use_system_libopenjpeg2 = true
 
   # Don't build against bundled libpng.
   use_system_libpng = false
