diff --git a/build/build_config.h b/build/build_config.h
index 35661b6..c4b5e87 100644
--- a/build/build_config.h
+++ b/build/build_config.h
@@ -100,6 +100,19 @@
 // We really are using glibc, not uClibc pretending to be glibc.
 #define LIBC_GLIBC 1
 #endif
+#if !defined(_GNU_SOURCE)
+#define _GNU_SOURCE
+#include <features.h>
+#if !defined(__USE_GNU)
+#define LIBC_MUSL 1
+#endif // !defined (__USE_GNU)
+#undef _GNU_SOURCE
+#else
+#include <features.h>
+#if !defined(__USE_GNU)
+#define LIBC_MUSL 1
+#endif // !defined(__USE_GNU)
+#endif // defined(_GNU_SOURCE)
 #elif defined(_WIN32)
 #define OS_WIN 1
 #elif defined(__Fuchsia__)
