--- a/chrome/browser/ui/lens/lens_overlay_request_id_generator.h
+++ b/chrome/browser/ui/lens/lens_overlay_request_id_generator.h
@@ -5,6 +5,7 @@
 #ifndef CHROME_BROWSER_UI_LENS_LENS_OVERLAY_REQUEST_ID_GENERATOR_H_
 #define CHROME_BROWSER_UI_LENS_LENS_OVERLAY_REQUEST_ID_GENERATOR_H_
 
+#include <optional>
 #include "third_party/lens_server_proto/lens_overlay_routing_info.pb.h"
 #include "third_party/lens_server_proto/lens_overlay_server.pb.h"
 #include "third_party/lens_server_proto/lens_overlay_service_deps.pb.h"

