--- a/third_party/electron_node/configure.orig	2021-12-24 08:11:32.465879983 +0100
+++ b/third_party/electron_node/configure	2021-12-24 08:11:53.152546320 +0100
@@ -21,7 +21,7 @@
   from distutils.spawn import find_executable as which
 
 print('Node.js configure: Found Python {}.{}.{}...'.format(*sys.version_info))
-acceptable_pythons = ((3, 9), (3, 8), (3, 7), (3, 6))
+acceptable_pythons = ((3, 10), (3, 9), (3, 8), (3, 7), (3, 6))
 if sys.version_info[:2] in acceptable_pythons:
   import configure
 else:
