diff --git a/components/paint_preview/common/subset_font.cc b/components/paint_preview/common/subset_font.cc
index 8ff0540d9a913..20a7d374741b6 100644
--- a/components/paint_preview/common/subset_font.cc
+++ b/components/paint_preview/common/subset_font.cc
@@ -72,9 +72,11 @@ sk_sp<SkData> SubsetFont(SkTypeface* typeface, const GlyphUsage& usage) {
   hb_set_t* glyphs =
       hb_subset_input_glyph_set(input.get());  // Owned by |input|.
   usage.ForEach(base::BindRepeating(&AddGlyphs, base::Unretained(glyphs)));
-  hb_subset_input_set_retain_gids(input.get(), true);
+  hb_subset_input_set_flags(input.get(), HB_SUBSET_FLAGS_RETAIN_GIDS);
 
-  HbScoped<hb_face_t> subset_face(hb_subset(face.get(), input.get()));
+  HbScoped<hb_face_t> subset_face(hb_subset_or_fail(face.get(), input.get()));
+  if (!subset_face)
+    return nullptr;
   HbScoped<hb_blob_t> subset_blob(hb_face_reference_blob(subset_face.get()));
   if (!subset_blob)
     return nullptr;
diff --git a/third_party/harfbuzz-ng/BUILD.gn b/third_party/harfbuzz-ng/BUILD.gn
index be6f93b56915a..45630bc80e4bb 100644
--- a/third_party/harfbuzz-ng/BUILD.gn
+++ b/third_party/harfbuzz-ng/BUILD.gn
@@ -108,6 +108,8 @@ if (use_system_harfbuzz) {
       "src/src/hb-map.cc",
       "src/src/hb-map.hh",
       "src/src/hb-meta.hh",
+      "src/src/hb-ms-feature-ranges.cc",
+      "src/src/hb-ms-feature-ranges.hh",
       "src/src/hb-mutex.hh",
       "src/src/hb-number-parser.hh",
       "src/src/hb-number.cc",
@@ -159,6 +161,7 @@ if (use_system_harfbuzz) {
       "src/src/hb-ot-name.cc",
       "src/src/hb-ot-os2-unicode-ranges.hh",
       "src/src/hb-ot-post-macroman.hh",
+      "src/src/hb-ot-post-table-v2subset.hh",
       "src/src/hb-ot-shape-complex-arabic-fallback.hh",
       "src/src/hb-ot-shape-complex-arabic-joining-list.hh",
       "src/src/hb-ot-shape-complex-arabic-table.hh",
diff --git a/third_party/harfbuzz-ng/fuzz/hb_subset_fuzzer.cc b/third_party/harfbuzz-ng/fuzz/hb_subset_fuzzer.cc
index 703de5d6a77f7..2e74e0b241b20 100644
--- a/third_party/harfbuzz-ng/fuzz/hb_subset_fuzzer.cc
+++ b/third_party/harfbuzz-ng/fuzz/hb_subset_fuzzer.cc
@@ -21,13 +21,14 @@ void TrySubset(hb_face_t* face,
                const hb_codepoint_t text[],
                const int text_length,
                const uint8_t flags) {
-  bool drop_hints = flags & (1 << 0);
   bool drop_layout = flags & (1 << 1);
-  bool retain_gids = flags & (1 << 2);
+  unsigned input_set_flags =
+      0                                                          //
+      | ((flags & (1 << 0)) ? HB_SUBSET_FLAGS_NO_HINTING : 0)    //
+      | ((flags & (1 << 2)) ? HB_SUBSET_FLAGS_RETAIN_GIDS : 0);  //
 
   HbScoped<hb_subset_input_t> input(hb_subset_input_create_or_fail());
-  hb_subset_input_set_drop_hints(input.get(), drop_hints);
-  hb_subset_input_set_retain_gids(input.get(), retain_gids);
+  hb_subset_input_set_flags(input.get(), input_set_flags);
   hb_set_t* codepoints = hb_subset_input_unicode_set(input.get());
 
   if (!drop_layout) {
@@ -43,7 +44,7 @@ void TrySubset(hb_face_t* face,
     hb_set_add(codepoints, text[i]);
   }
 
-  HbScoped<hb_face_t> result(hb_subset(face, input.get()));
+  HbScoped<hb_face_t> result(hb_subset_or_fail(face, input.get()));
   HbScoped<hb_blob_t> blob(hb_face_reference_blob(result.get()));
   uint32_t length;
   const char* data = hb_blob_get_data(blob.get(), &length);
