--- a/chrome/browser/download/BUILD.gn
+++ b/chrome/browser/download/BUILD.gn
@@ -574,9 +574,6 @@ if (!is_android) {
       "//components/infobars/content",
       "//components/lookalikes/core:safety_tips",
       "//components/page_load_metrics/browser:test_support",
-      "//components/safe_browsing/content/browser:safe_browsing_service",
-      "//components/safe_browsing/content/common:file_type_policies",
-      "//components/safe_browsing/content/common:file_type_policies_test_support",
       "//components/services/quarantine:test_support",
       "//content/public/browser",
       "//content/test:test_support",
--- a/chrome/browser/feedback/BUILD.gn
+++ b/chrome/browser/feedback/BUILD.gn
@@ -140,7 +140,6 @@ source_set("feedback_impl") {
       "//components/device_event_log",
       "//components/feedback",
       "//components/performance_manager",
-      "//components/safe_browsing/content/browser:client_side_detection",
       "//components/supervised_user/core/browser",
       "//extensions/browser",
       "//extensions/browser/api/feedback_private",
@@ -274,7 +273,6 @@ if (!is_android) {
       "//chrome/browser/ui/toolbar",
       "//chrome/test:test_support",
       "//chrome/test:test_support_ui",
-      "//components/safe_browsing/content/browser:client_side_detection",
       "//content/test:test_support",
       "//testing/gtest",
     ]
--- a/chrome/browser/ui/views/file_system_access/BUILD.gn
+++ b/chrome/browser/ui/views/file_system_access/BUILD.gn
@@ -97,7 +97,6 @@ if (!is_android) {
         "//chrome/browser/safe_browsing",
         "//chrome/browser/ui/test:dialog",
         "//chrome/browser/ui/views:file_system_access_test_support",
-        "//components/safe_browsing/content/common:file_type_policies_test_support",
       ]
     }
   }
--- a/chrome/browser/ui/webui/feedback/report_unsafe_site/BUILD.gn
+++ b/chrome/browser/ui/webui/feedback/report_unsafe_site/BUILD.gn
@@ -26,7 +26,6 @@ static_library("report_unsafe_site") {
     "//chrome/browser/feedback/report_unsafe_site",
     "//chrome/browser/ui/webui/top_chrome",
     "//chrome/common",
-    "//components/safe_browsing/content/browser:client_side_detection",
     "//components/url_formatter",
     "//components/viz/common",
     "//content/public/browser",
--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -2249,7 +2249,6 @@ if (!is_android) {
       "//chrome/browser/renderer_host:browser_tests",
       "//chrome/browser/resource_coordinator:mojo_bindings",
       "//chrome/browser/safe_browsing:advanced_protection",
-      "//chrome/browser/safe_browsing:blocking_page_platform_test_support",
       "//chrome/browser/safe_browsing:verdict_cache_manager_factory",
       "//chrome/browser/search",
       "//chrome/browser/search:browser_tests",
@@ -2682,7 +2681,6 @@ if (!is_android) {
       "//components/resources",
       "//components/safe_browsing:buildflags",
       "//components/safe_browsing/content/browser",
-      "//components/safe_browsing/content/browser:client_side_detection_images_cache",
       "//components/safe_browsing/content/browser:safe_browsing_service",
       "//components/safe_browsing/content/browser/password_protection",
       "//components/safe_browsing/content/browser/password_protection:test_support",
@@ -5716,7 +5714,6 @@ test("unit_tests") {
     "//components/resources",
     "//components/safe_browsing:buildflags",
     "//components/safe_browsing/content/browser",
-    "//components/safe_browsing/content/browser:client_side_detection_images_cache",
     "//components/safe_browsing/content/browser/password_protection",
     "//components/safe_browsing/content/browser/password_protection:mock_password_protection",
     "//components/safe_browsing/content/browser/web_ui",
@@ -8791,7 +8788,6 @@ if (!is_android && !is_chromeos_device)
       "//components/privacy_sandbox:tracking_protection_settings",
       "//components/reading_list/core",
       "//components/resources",
-      "//components/safe_browsing/content/browser:client_side_detection",
       "//components/safe_browsing/core/common",
       "//components/safe_browsing/core/common:safe_browsing_policy_handler",
       "//components/search",
