--- chromium-97.0.4692.71/media/filters/ffmpeg_demuxer.cc
+++ chromium-97.0.4692.71/media/filters/ffmpeg_demuxer.cc
@@ -1610,7 +1610,7 @@
   for (const auto& stream : streams_) {
     if (!stream || stream->IsEnabled() != enabled)
       continue;
-    if (av_stream_get_first_dts(stream->av_stream()) == kInvalidPTSMarker)
+    if ((stream->av_stream()->first_dts) == kInvalidPTSMarker)
       continue;
     if (!lowest_start_time_stream ||
         stream->start_time() < lowest_start_time_stream->start_time()) {
@@ -1631,7 +1631,7 @@
     if (stream->type() != DemuxerStream::VIDEO)
       continue;
 
-    if (av_stream_get_first_dts(stream->av_stream()) == kInvalidPTSMarker)
+    if ((stream->av_stream()->first_dts) == kInvalidPTSMarker)
       continue;
 
     if (!stream->IsEnabled())

