diff -up chromium-99.0.4818.0/media/audio/audio_opus_encoder.h.omv~ chromium-99.0.4818.0/media/audio/audio_opus_encoder.h
--- chromium-99.0.4818.0/media/audio/audio_opus_encoder.h.omv~	2022-01-13 05:42:57.571250471 +0100
+++ chromium-99.0.4818.0/media/audio/audio_opus_encoder.h	2022-01-13 05:43:06.233291236 +0100
@@ -13,7 +13,7 @@
 #include "media/base/audio_encoder.h"
 #include "media/base/audio_push_fifo.h"
 #include "media/base/audio_timestamp_helper.h"
-#include "third_party/opus/src/include/opus.h"
+#include <opus/opus.h>
 
 namespace media {
 
--- a/media/ffmpeg/ffmpeg_common.cc
+++ b/media/ffmpeg/ffmpeg_common.cc
@@ -1025,7 +1025,7 @@
   static const base::NoDestructor<std::string> kAllowedAudioCodecs([]() {
     // This should match the configured lists in //third_party/ffmpeg.
     std::string allowed_decoders(
-        "vorbis,libopus,flac,pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,"
+        "vorbis,opus,flac,pcm_u8,pcm_s16le,pcm_s24le,pcm_s32le,pcm_f32le,"
         "mp3,pcm_s16be,pcm_s24be,pcm_mulaw,pcm_alaw");
 #if BUILDFLAG(USE_PROPRIETARY_CODECS)
     allowed_decoders += ",aac";

