--- a/chrome/test/BUILD.gn
+++ b/chrome/test/BUILD.gn
@@ -8245,12 +8245,6 @@
 
     data += [ "//ash/components/arc/test/data/icons/" ]
 
-    data_deps += [
-      # enclave_manager_unittest.cc runs this binary as part of its testing
-      # process.
-      "//third_party/cloud_authenticator/test/local_service:cloud_authenticator_test_service",
-    ]
-
     if (include_js2gtest_tests && is_chromeos_ash) {
       data += js2gtest_js_libraries
       deps +=
