From cff31bc6836b69b1b6b67a29a264dc7d763f76a4 Mon Sep 17 00:00:00 2001
From: Matt Jolly <kangie@gentoo.org>
Date: Tue, 9 Sep 2025 21:22:40 +1000
Subject: [PATCH] update compiler patch for 142

Signed-off-by: Matt Jolly <kangie@gentoo.org>
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -357,9 +357,6 @@ config("compiler") {
 
   configs += [
     # See the definitions below.
-    ":clang_revision",
-    ":rustc_revision",
-    ":compiler_cpu_abi",
     ":compiler_codegen",
     ":compiler_deterministic",
     ":clang_warning_suppression",
@@ -1659,43 +1656,6 @@ config("compiler_deterministic") {
     }
   }
 
-  # Makes builds independent of absolute file path.
-  if (is_clang && strip_absolute_paths_from_debug_symbols) {
-    # If debug option is given, clang includes $cwd in debug info by default.
-    # For such build, this flag generates reproducible obj files even we use
-    # different build directory like "out/feature_a" and "out/feature_b" if
-    # we build same files with same compile flag.
-    # Other paths are already given in relative, no need to normalize them.
-
-    # -ffile-compilation-dir is an alias for both -fdebug-compilation-dir=
-    # and -fcoverage-compilation-dir=.
-    cflags += [ "-ffile-compilation-dir=." ]
-
-    # Convert absolute paths to relative paths. Expands to, for example:
-    #   -file-prefix-map /path/to/chromium/src=../..
-    swiftflags += [
-      "-file-prefix-map",
-      rebase_path("//.", "") + "=" + rebase_path("//.", root_build_dir),
-    ]
-    if (!is_win) {
-      # We don't use clang -cc1as on Windows (yet? https://crbug.com/762167)
-      asmflags = [ "-Wa,-fdebug-compilation-dir,." ]
-    }
-
-    if (is_win && use_lld) {
-      if (symbol_level == 2 || (is_clang && using_sanitizer)) {
-        # Absolutize source file paths for PDB. Pass the real build directory
-        # if the pdb contains source-level debug information and if linker
-        # reproducibility is not critical.
-        ldflags += [ "/PDBSourcePath:" + rebase_path(root_build_dir) ]
-      } else {
-        # Use a fake fixed base directory for paths in the pdb to make the pdb
-        # output fully deterministic and independent of the build directory.
-        ldflags += [ "/PDBSourcePath:o:\fake\prefix" ]
-      }
-    }
-  }
-
   # Tells the compiler not to use absolute paths when passing the default
   # paths to the tools it invokes. We don't want this because we don't
   # really need it and it can mess up the RBE cache entries.
@@ -1728,88 +1688,6 @@ config("compiler_deterministic") {
   }
 }
 
-config("clang_revision") {
-  if (is_clang && clang_base_path == default_clang_base_path &&
-      current_os != "zos") {
-    _perform_consistency_checks = current_toolchain == default_toolchain
-    if (llvm_force_head_revision) {
-      _head_revision_stamp_path = "//third_party/llvm-build/force_head_revision"
-      _head_revision = ""
-      if (path_exists(_head_revision_stamp_path)) {
-        _head_revision = read_file(_head_revision_stamp_path, "trim string")
-      }
-      assert(
-          _head_revision != "",
-          "llvm_force_head_revision=true, but no locally built version was detected.")
-      _clang_revision = _head_revision
-    } else {
-      _clang_revision = read_file(
-              "//third_party/llvm-build/Release+Asserts/cr_build_revision",
-              "trim string")
-
-      # Ensure that the synced clang version matches what's in git.
-      if (_perform_consistency_checks) {
-        # Parse the clang version from the Python script.
-        _clang_version_lines = filter_include(
-                read_file("//tools/clang/scripts/update.py", "list lines"),
-                [ "CLANG_*REVISION = *" ])
-        _py_revision =
-            string_replace(_clang_version_lines[0], "CLANG_REVISION = '", "")
-        _py_revision = string_replace(_py_revision, "'", "")
-        _py_subrevision =
-            string_replace(_clang_version_lines[1], "CLANG_SUB_REVISION = ", "")
-        _expected_clang_revision = "$_py_revision-$_py_subrevision"
-
-        # TODO(agrieve): Change filter_include to _clang_revision ==
-        # _expected_clang_revision once angle & webrtc use a GCS DEPS entry
-        # rather than a update.py runhook.
-        assert(
-            filter_include([ _clang_revision ],
-                           [
-                             _expected_clang_revision,
-                             "$_expected_clang_revision,*",
-                           ]) != [],
-            "clang_revision=\"$_clang_revision\" but update.py expected \"$_expected_clang_revision\". Did you forget to gclient sync?")
-      }
-    }
-
-    if (_perform_consistency_checks) {
-      # Ensure that the revision matches the version major expected by GN.
-      _versions_match = filter_include([ _clang_revision ],
-                                       [ "llvmorg-$clang_version-*" ]) != []
-      assert(
-          _versions_match,
-          "clang_revision=\"$_clang_revision\" but clang_version=\"$clang_version\". clang_version in build/toolchain/toolchain.gni is likely outdated.")
-    }
-
-    if (toolchain_has_rust && _perform_consistency_checks &&
-        !rust_force_head_revision) {
-      # Ensure that the synced rust version matches what's in git.
-      _rust_revision_lines =
-          filter_include(read_file("//tools/rust/update_rust.py", "list lines"),
-                         [ "RUST_*REVISION = *" ])
-      _py_revision =
-          string_replace(_rust_revision_lines[0], "RUST_REVISION = '", "")
-      _py_revision = string_replace(_py_revision, "'", "")
-      _py_subrevision =
-          string_replace(_rust_revision_lines[1], "RUST_SUB_REVISION = ", "")
-      _expected_rust_revision = "$_py_revision-$_py_subrevision"
-
-      # Ensure the rust version matches the clang version.
-      assert(
-          filter_include([ rustc_revision ],
-                         [ "*-$_expected_rust_revision-*" ]) != [],
-          "rustc_revision=\"$rustc_revision\" but update_rust.py expected \"$_expected_rust_revision\". Run \"gclient sync\"?")
-    }
-
-    # This is here so that all files get recompiled after a clang roll and
-    # when turning clang on or off. (defines are passed via the command line,
-    # and build system rebuild things when their commandline changes). Nothing
-    # should ever read this define.
-    defines = [ "CR_CLANG_REVISION=\"$_clang_revision\"" ]
-  }
-}
-
 # Controls the usage of a warning suppression mapping (WSM) file to suppress
 # warnings based on the path of the file they come from. It's controlled by the
 # `clang_warning_suppression_file` gn argument , which points to a text file
@@ -1853,18 +1731,6 @@ config("sanitize_c_array_bounds") {
   }
 }
 
-config("rustc_revision") {
-  if (rustc_revision != "") {
-    # Similar to the above config, this is here so that all files get recompiled
-    # after a rustc roll. Nothing should ever read this cfg. This will not be
-    # set if a custom toolchain is used.
-    rustflags = [
-      "--cfg",
-      "cr_rustc_revision=\"$rustc_revision\"",
-    ]
-  }
-}
-
 config("compiler_arm_fpu") {
   if (current_cpu == "arm" && !is_ios) {
     cflags = [ "-mfpu=$arm_fpu" ]
@@ -2330,11 +2196,7 @@ config("chromium_code") {
       defines = [ "_HAS_NODISCARD" ]
     }
   } else {
-    cflags = [ "-Wall" ]
-    if (is_clang) {
-      # Enable extra warnings for chromium_code when we control the compiler.
-      cflags += [ "-Wextra" ]
-    }
+    cflags = []
 
     # In Chromium code, we define __STDC_foo_MACROS in order to get the
     # C99 macros on Mac and Linux.
@@ -2343,23 +2205,6 @@ config("chromium_code") {
       "__STDC_FORMAT_MACROS",
     ]
 
-    if (!is_debug && !using_sanitizer && current_cpu != "s390x" &&
-        current_cpu != "ppc64" && current_cpu != "mips" &&
-        current_cpu != "mips64" && current_cpu != "riscv64" &&
-        current_cpu != "loong64") {
-      # Non-chromium code is not guaranteed to compile cleanly with
-      # _FORTIFY_SOURCE. Also, fortified build may fail when optimizations are
-      # disabled, so only do that for Release build.
-      fortify_level = "2"
-
-      # ChromeOS's toolchain supports a high-quality _FORTIFY_SOURCE=3
-      # implementation with a few custom glibc patches. Use that if it's
-      # available.
-      if (is_chromeos_device) {
-        fortify_level = "3"
-      }
-      defines += [ "_FORTIFY_SOURCE=" + fortify_level ]
-    }
 
     if (is_apple) {
       cflags_objc = [ "-Wimplicit-retain-self" ]
@@ -2784,7 +2629,8 @@ config("default_stack_frames") {
 # [0]: https://pinpoint-dot-chromeperf.appspot.com/job/147634a8be0000
 # [1]: https://pinpoint-dot-chromeperf.appspot.com/job/132bc772be0000
 # [2]: https://crrev.com/c/5447532
-config("optimize") {
+config("optimize") { }
+config("xoptimize") {
   if (is_win) {
     # clang-cl's /O2 corresponds to clang's -O3, and really want -O2 for
     # consistency with the other platforms.
@@ -2833,7 +2679,8 @@ config("optimize") {
 }
 
 # Turn off optimizations.
-config("no_optimize") {
+config("no_optimize") { }
+config("xno_optimize") {
   if (is_win) {
     cflags = [
       "/Od",  # Disable optimization.
@@ -2868,7 +2715,8 @@ config("no_optimize") {
 # Turns up the optimization level. Used to explicitly enable -O2 instead of
 # -Os for select targets on platforms that use optimize_for_size. No-op
 # elsewhere.
-config("optimize_max") {
+config("optimize_max") { }
+config("xoptimize_max") {
   ldflags = common_optimize_on_ldflags
   if (is_win) {
     # Favor speed over size, /O2 must be before the common flags.
@@ -2889,7 +2737,8 @@ config("optimize_max") {
 #
 # TODO(crbug.com/41259697) - rework how all of these configs are related
 # so that we don't need this disclaimer.
-config("optimize_speed") {
+config("optimize_speed") { }
+config("xoptimize_speed") {
   ldflags = common_optimize_on_ldflags
   if (is_win) {
     # Favor speed over size, /O2 must be before the common flags.
@@ -2906,7 +2755,8 @@ config("optimize_speed") {
   rustflags = [ "-Copt-level=3" ]
 }
 
-config("optimize_fuzzing") {
+config("optimize_fuzzing") { }
+config("xoptimize_fuzzing") {
   cflags = [ "-O1" ] + common_optimize_on_cflags
   rustflags = [ "-Copt-level=1" ]
   ldflags = common_optimize_on_ldflags
@@ -3033,7 +2883,8 @@ config("win_pdbaltpath") {
 }
 
 # Full symbols.
-config("symbols") {
+config("symbols") { }
+config("xsymbols") {
   rustflags = []
   configs = []
   if (is_win) {
@@ -3198,7 +3049,8 @@ config("symbols") {
 # Minimal symbols.
 # This config guarantees to hold symbol for stack trace which are shown to user
 # when crash happens in unittests running on buildbot.
-config("minimal_symbols") {
+config("minimal_symbols") { }
+config("xminimal_symbols") {
   rustflags = []
   if (is_win) {
     # Functions, files, and line tables only.
@@ -3274,7 +3126,8 @@ config("minimal_symbols") {
 # This configuration contains function names only. That is, the compiler is
 # told to not generate debug information and the linker then just puts function
 # names in the final debug information.
-config("no_symbols") {
+config("no_symbols") { }
+config("xno_symbols") {
   if (is_win) {
     ldflags = [ "/DEBUG" ]
 
