diff -up chromium-92.0.4515.107/third_party/jinja2/tests.py.py3 chromium-92.0.4515.107/third_party/jinja2/tests.py
--- chromium-92.0.4515.107/third_party/jinja2/tests.py.py3	2021-07-28 15:53:45.670961029 -0400
+++ chromium-92.0.4515.107/third_party/jinja2/tests.py	2021-07-28 15:55:56.637013096 -0400
@@ -10,7 +10,12 @@
 """
 import operator
 import re
-from collections import Mapping
+import sys
+if sys.version_info[:2] >= (3, 8):  # pragma: no cover
+    from collections.abc import Mapping
+else:  # pragma: no cover
+    from collections import Mapping
+
 from jinja2.runtime import Undefined
 from jinja2._compat import text_type, string_types, integer_types
 import decimal
