--- a/skia/BUILD.gn
+++ b/skia/BUILD.gn
@@ -8,7 +8,6 @@
 import("//build/config/freetype/freetype.gni")
 import("//build/config/rust.gni")
 import("//build/config/sanitizers/sanitizers.gni")
-import("//build/rust/rust_static_library.gni")
 import("//gpu/vulkan/features.gni")
 import("//testing/test.gni")
 import("//third_party/skia/gn/shared_sources.gni")
@@ -28,28 +27,6 @@
     "SKIA_SUPPORT_SKOTTIE=$skia_support_skottie",
     "SKIA_USE_DAWN=$skia_use_dawn",
     "SKIA_USE_METAL=$skia_use_metal",
-  ]
-}
-
-source_set("path_bridge") {
-  sources = [ "//third_party/skia/src/ports/fontations/src/skpath_bridge.h" ]
-}
-
-rust_static_library("bridge_rust_side") {
-  allow_unsafe = true
-  crate_root = "//third_party/skia/src/ports/fontations/src/ffi.rs"
-  sources = [ "//third_party/skia/src/ports/fontations/src/ffi.rs" ]
-  cxx_bindings = [ "//third_party/skia/src/ports/fontations/src/ffi.rs" ]
-  deps = [
-    ":path_bridge",
-    "//third_party/rust/font_types/v0_5:lib",
-    "//third_party/rust/read_fonts/v0_19:lib",
-    "//third_party/rust/skrifa/v0_19:lib",
-  ]
-  configs -= [ "//build/config/compiler:chromium_code" ]
-  configs += [
-    ":skia_config",
-    "//build/config/compiler:no_chromium_code",
   ]
 }
 
@@ -400,14 +377,6 @@
       "//third_party/skia/src/ports/SkFontHost_FreeType_common.cpp",
       "//third_party/skia/src/ports/SkFontMgr_custom.cpp",
       "//third_party/skia/src/ports/SkFontMgr_custom_empty.cpp",
-      "//third_party/skia/src/ports/SkTypeface_fontations.cpp",
-      "//third_party/skia/src/ports/SkTypeface_fontations_priv.h",
-    ]
-
-    # Fontations Rust/C++ bridge interfaces.
-    deps += [
-      ":bridge_rust_side",
-      ":path_bridge",
     ]
   }
 
--- a/third_party/blink/renderer/platform/fonts/web_font_typeface_factory.cc
+++ b/third_party/blink/renderer/platform/fonts/web_font_typeface_factory.cc
@@ -12,7 +12,6 @@
 #include "third_party/blink/renderer/platform/fonts/opentype/font_format_check.h"
 #include "third_party/skia/include/core/SkStream.h"
 #include "third_party/skia/include/core/SkTypeface.h"
-#include "third_party/skia/include/ports/SkTypeface_fontations.h"
 
 #if BUILDFLAG(IS_WIN)
 #include "third_party/blink/renderer/platform/fonts/win/dwrite_font_format_support.h"
@@ -30,12 +29,6 @@
 namespace {
 
 sk_sp<SkTypeface> MakeTypefaceDefaultFontMgr(sk_sp<SkData> data) {
-#if !(BUILDFLAG(IS_WIN) || BUILDFLAG(IS_APPLE))
-  if (RuntimeEnabledFeatures::FontationsFontBackendEnabled()) {
-    std::unique_ptr<SkStreamAsset> stream(new SkMemoryStream(data));
-    return SkTypeface_Make_Fontations(std::move(stream), SkFontArguments());
-  }
-#endif
 
   sk_sp<SkFontMgr> font_manager;
 #if BUILDFLAG(IS_WIN)

