--- a/media/base/agtm.cc
+++ b/media/base/agtm.cc
@@ -38,6 +38,8 @@
   return t35_payload_without_country_code.subspan(prefix.size());
 }
 
+}  // namespace
+
 std::optional<base::span<const uint8_t>> GetAgtmFromT35(
     base::span<const uint8_t> t35_payload) {
   auto expected_prefix = base::span(kAgtmPrefix);
@@ -50,8 +52,6 @@
   }
   return t35_payload.subspan(prefix.size());
 }
-
-}  // namespace
 
 void SetAgtmFromT35WithCountryCode(
     gfx::HDRMetadata& hdr_metadata,
--- a/media/base/agtm.h
+++ b/media/base/agtm.h
@@ -28,6 +28,8 @@
 // metadata in `hdr_metadata`.
 MEDIA_EXPORT void SetAgtmFromT35(gfx::HDRMetadata& hdr_metadata,
                                  base::span<const uint8_t> t35_payload);
+MEDIA_EXPORT std::optional<base::span<const uint8_t>> GetAgtmFromT35(
+    base::span<const uint8_t> t35_payload);
 
 }  // namespace media
 

