From 307a0f63dd9b118f4b8470ed3d7567e81fdb7a6d Mon Sep 17 00:00:00 2001
From: Mike Gilbert <floppym@gentoo.org>
Date: Fri, 22 Apr 2022 09:05:24 +0000
Subject: [PATCH] Disable various compiler configs

---
 build/config/compiler/BUILD.gn | 114 +++++----------------------------
 1 file changed, 17 insertions(+), 97 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index d40843b..b92f03b 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1538,6 +1472,15 @@ config("default_warnings") {
         cflags += [ "-Wno-deprecated-non-prototype" ]
       }
 
+      if (!is_nacl && !(is_chromeos ||
+                        default_toolchain == "//build/toolchain/cros:target")) {
+        # TODO(https://crbug.com/1316298): Re-enable once test failure is figured out
+        cflags += [
+          "-Xclang",
+          "-no-opaque-pointers",
+        ]
+      }
+
       if (is_fuchsia) {
         # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
         cflags += [ "-Wno-deprecated-copy" ]
