--- a/build/config/locales.gni
+++ b/build/config/locales.gni
@@ -166,12 +166,6 @@
 
 # Setup |platform_pak_locales| for each platform.
 platform_pak_locales = all_chrome_locales
-if (is_android) {
-  platform_pak_locales -= _non_android_locales
-  extended_locales -= _non_android_locales
-} else {
-  platform_pak_locales -= extended_locales
-}
 
 # The base list for all platforms except Android excludes the extended locales.
 # Add or subtract platform specific locales below.
--- a/chrome/app/resources/locale_settings_linux.grd
+++ b/chrome/app/resources/locale_settings_linux.grd
@@ -4,7 +4,7 @@
     <output filename="grit/platform_locale_settings.h" type="rc_header">
       <emit emit_type='prepend'></emit>
     </output>
-    <if expr="is_android">
+    <if expr="is_linux">
       <output filename="platform_locale_settings_as.pak" type="data_package" lang="as" />
       <output filename="platform_locale_settings_az.pak" type="data_package" lang="az" />
       <output filename="platform_locale_settings_be.pak" type="data_package" lang="be" />
@@ -38,6 +38,7 @@
     <output filename="platform_locale_settings_bg.pak" type="data_package" lang="bg" />
     <output filename="platform_locale_settings_bn.pak" type="data_package" lang="bn" />
     <output filename="platform_locale_settings_ca.pak" type="data_package" lang="ca" />
+    <output filename="platform_locale_settings_cy.pak" type="data_package" lang="cy" />
     <output filename="platform_locale_settings_cs.pak" type="data_package" lang="cs" />
     <output filename="platform_locale_settings_da.pak" type="data_package" lang="da" />
     <output filename="platform_locale_settings_de.pak" type="data_package" lang="de" />

