diff -ru chromium-121.0.6167.160.orig/chrome/browser/performance_manager/policies/probability_distribution.h chromium-121.0.6167.160/chrome/browser/performance_manager/policies/probability_distribution.h
--- chromium-121.0.6167.160.orig/chrome/browser/performance_manager/policies/probability_distribution.h	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/chrome/browser/performance_manager/policies/probability_distribution.h	2024-02-10 00:00:00.000000000 +0000
@@ -5,6 +5,7 @@
 #ifndef CHROME_BROWSER_PERFORMANCE_MANAGER_POLICIES_PROBABILITY_DISTRIBUTION_H_
 #define CHROME_BROWSER_PERFORMANCE_MANAGER_POLICIES_PROBABILITY_DISTRIBUTION_H_
 
+#include <cstdint>
 #include <utility>
 #include <vector>
 
diff -ru chromium-121.0.6167.160.orig/mojo/public/cpp/bindings/type_converter.h chromium-121.0.6167.160/mojo/public/cpp/bindings/type_converter.h
--- chromium-121.0.6167.160.orig/mojo/public/cpp/bindings/type_converter.h	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/mojo/public/cpp/bindings/type_converter.h	2024-02-10 00:00:00.000000000 +0000
@@ -98,16 +98,7 @@
 template <typename T, typename U>
   requires(!std::is_pointer_v<U>)
 inline T ConvertTo(const U& obj) {
-  if constexpr (requires {
-                  {
-                    mojo::ConvertTo<T>(std::to_address(obj))
-
-                  } -> std::same_as<T>;
-                }) {
-    return mojo::ConvertTo<T>(std::to_address(obj));
-  } else {
     return TypeConverter<T, U>::Convert(obj);
-  }
 }
 
 template <typename T>
diff -ru chromium-121.0.6167.160.orig/third_party/blink/renderer/modules/handwriting/handwriting_type_converters.cc chromium-121.0.6167.160/third_party/blink/renderer/modules/handwriting/handwriting_type_converters.cc
--- chromium-121.0.6167.160.orig/third_party/blink/renderer/modules/handwriting/handwriting_type_converters.cc	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/third_party/blink/renderer/modules/handwriting/handwriting_type_converters.cc	2024-02-10 00:00:00.000000000 +0000
@@ -19,6 +19,8 @@
 #include "third_party/blink/renderer/modules/handwriting/handwriting_stroke.h"
 #include "third_party/blink/renderer/modules/modules_export.h"
 
+#include <memory>
+
 namespace mojo {
 
 using handwriting::mojom::blink::HandwritingDrawingSegmentPtr;
@@ -45,6 +47,11 @@
   return output;
 }
 
+template <>
+inline HandwritingPointPtr ConvertTo(const cppgc::internal::BasicMember<const blink::HandwritingPoint, cppgc::internal::StrongMemberTag, cppgc::internal::DijkstraWriteBarrierPolicy> &obj) {
+    return mojo::ConvertTo<HandwritingPointPtr>(std::to_address(obj));
+}
+
 // static
 HandwritingStrokePtr
 TypeConverter<HandwritingStrokePtr, blink::HandwritingStroke*>::Convert(
diff -ru chromium-121.0.6167.160.orig/third_party/blink/renderer/modules/payments/secure_payment_confirmation_type_converter.cc chromium-121.0.6167.160/third_party/blink/renderer/modules/payments/secure_payment_confirmation_type_converter.cc
--- chromium-121.0.6167.160.orig/third_party/blink/renderer/modules/payments/secure_payment_confirmation_type_converter.cc	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/third_party/blink/renderer/modules/payments/secure_payment_confirmation_type_converter.cc	2024-02-10 00:00:00.000000000 +0000
@@ -17,6 +17,14 @@
 
 namespace mojo {
 
+template <>
+inline 
+Vector<uint8_t> ConvertTo(const
+      cppgc::internal::BasicMember<blink::V8UnionArrayBufferOrArrayBufferView, cppgc::internal::StrongMemberTag, cppgc::internal::DijkstraWriteBarrierPolicy> &obj) {
+    return mojo::ConvertTo< Vector<uint8_t> >(std::to_address(obj));
+} 
+
+
 payments::mojom::blink::SecurePaymentConfirmationRequestPtr
 TypeConverter<payments::mojom::blink::SecurePaymentConfirmationRequestPtr,
               blink::SecurePaymentConfirmationRequest*>::
diff -ru chromium-121.0.6167.160.orig/third_party/blink/renderer/platform/fonts/simple_font_data.h chromium-121.0.6167.160/third_party/blink/renderer/platform/fonts/simple_font_data.h
--- chromium-121.0.6167.160.orig/third_party/blink/renderer/platform/fonts/simple_font_data.h	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/third_party/blink/renderer/platform/fonts/simple_font_data.h	2024-02-10 00:00:00.000000000 +0000
@@ -25,6 +25,7 @@
 #define THIRD_PARTY_BLINK_RENDERER_PLATFORM_FONTS_SIMPLE_FONT_DATA_H_
 
 #include <memory>
+#include <mutex>
 #include <utility>
 
 #include "build/build_config.h"
diff -ru chromium-121.0.6167.160.orig/ui/display/types/display_color_management.h chromium-121.0.6167.160/ui/display/types/display_color_management.h
--- chromium-121.0.6167.160.orig/ui/display/types/display_color_management.h	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/ui/display/types/display_color_management.h	2024-02-09 00:00:00.000000000 +0000
@@ -5,6 +5,7 @@
 #ifndef UI_DISPLAY_TYPES_DISPLAY_COLOR_MANAGEMENT_H_
 #define UI_DISPLAY_TYPES_DISPLAY_COLOR_MANAGEMENT_H_
 
+#include <string>
 #include <vector>
 
 #include "third_party/skia/modules/skcms/skcms.h"
diff -ru chromium-121.0.6167.160.orig/ui/gfx/x/visual_manager.cc chromium-121.0.6167.160/ui/gfx/x/visual_manager.cc
--- chromium-121.0.6167.160.orig/ui/gfx/x/visual_manager.cc	2024-02-07 00:00:00.000000000 +0000
+++ chromium-121.0.6167.160/ui/gfx/x/visual_manager.cc	2024-02-09 00:00:00.000000000 +0000
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <bitset>
 #include "ui/gfx/x/visual_manager.h"
 
 #include "base/strings/string_number_conversions.h"
