diff -ruN a/external_bin/gsutil/gsutil_5.35/gsutil/gsutil.py b/external_bin/gsutil/gsutil_5.35/gsutil/gsutil.py
--- a/external_bin/gsutil/gsutil_5.35/gsutil/gsutil.py
+++ b/external_bin/gsutil/gsutil_5.35/gsutil/gsutil.py
@@ -26,16 +26,6 @@
 
 # TODO: gsutil-beta: Distribute a pylint rc file.
 
-ver = sys.version_info
-if ver.major != 3 or ver.minor < 9 or ver.minor > 13:
-  sys.exit(
-    "Error: gsutil requires Python version 3.9-3.13, but a different version is installed.\n"
-    "You are currently running Python {}.{}\n"
-    "Follow the steps below to resolve this issue:\n"
-    "\t1. Switch to Python 3.9-3.13 using your Python version manager or install an appropriate version.\n"
-    "\t2. If you are unsure how to manage Python versions, visit [https://cloud.google.com/storage/docs/gsutil_install#specifications] for detailed instructions.".format(ver.major, ver.minor)
-  )
-
 # setup a string to load the correct httplib2
 if sys.version_info.major == 2:
   submodule_pyvers = 'python2'
