--- a/chrome/browser/chrome_browser_main.cc
+++ b/chrome/browser/chrome_browser_main.cc
@@ -503,8 +503,8 @@
       command_line, current_directory, startup_profile_path_info);
 
   // Record now as the last successful chrome start.
-  if (ShouldRecordActiveUse(command_line))
-    GoogleUpdateSettings::SetLastRunTime();
+  //if (ShouldRecordActiveUse(command_line))
+  //  GoogleUpdateSettings::SetLastRunTime();
 }
 #endif  // BUILDFLAG(ENABLE_PROCESS_SINGLETON)
 
@@ -1837,8 +1837,8 @@
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS_LACROS)
 
     // Record now as the last successful chrome start.
-    if (ShouldRecordActiveUse(*base::CommandLine::ForCurrentProcess()))
-      GoogleUpdateSettings::SetLastRunTime();
+    //if (ShouldRecordActiveUse(*base::CommandLine::ForCurrentProcess()))
+    //  GoogleUpdateSettings::SetLastRunTime();
 
     // Create the RunLoop for MainMessageLoopRun() to use and transfer
     // ownership of the browser's lifetime to the BrowserProcess.
--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -134,6 +134,7 @@
     "//build:chromeos_buildflags",
     "//chrome/common:buildflags",
     "//media:media_buildflags",
+    "//services/network/public/mojom",
   ]
 }
 
