--- a/chrome/browser/BUILD.gn
+++ b/chrome/browser/BUILD.gn
@@ -134,6 +134,7 @@
     "//build:chromeos_buildflags",
     "//chrome/common:buildflags",
     "//media:media_buildflags",
+    "//services/network/public/mojom",
   ]
 }
 
--- a/media/webrtc/BUILD.gn
+++ b/media/webrtc/BUILD.gn
@@ -26,7 +26,6 @@ component("webrtc") {
   deps = [
     "//base",
     "//build:chromecast_buildflags",
-    "//components/optimization_guide/core/inference:op_resolver",
     "//media",
     "//third_party/webrtc_overrides:webrtc_component",
   ]
--- a/media/webrtc/helpers.cc
+++ b/media/webrtc/helpers.cc
@@ -158,23 +158,6 @@ CreateWebRtcAudioProcessingModule(
       webrtc::EchoCanceller3Config::CreateDefaultMultichannelConfig();
   std::unique_ptr<webrtc::NeuralResidualEchoEstimator> echo_estimator;

-  // Fuchsia does not use the optimization guide.
-  // Avoid linking the op resolver to keep Fuchsia binary size down.
-  // TODO(crbug.com/450466837): Investigate if this build guard can be avoided.
-#if !BUILDFLAG(IS_FUCHSIA)
-  if (residual_echo_estimator_model) {
-    optimization_guide::TFLiteOpResolver op_resolver;
-    echo_estimator = webrtc::CreateNeuralResidualEchoEstimator(
-        residual_echo_estimator_model, &op_resolver);
-    if (echo_estimator) {
-      aec3_config = echo_estimator->GetConfiguration(/*multi_channel=*/false);
-      multichannel_aec3_config =
-          echo_estimator->GetConfiguration(/*multi_channel=*/true);
-    } else {
-      LOG(ERROR) << "Failed to initialize neural residual echo estimator.";
-    }
-  }
-#endif  // !BUILDFLAG(IS_FUCHSIA)

 #if BUILDFLAG(SYSTEM_LOOPBACK_AS_AEC_REFERENCE)
   if (settings.use_loopback_aec_reference) {
