--- a/third_party/blink/renderer/platform/image-decoders/BUILD.gn
+++ b/third_party/blink/renderer/platform/image-decoders/BUILD.gn
@@ -78,11 +78,10 @@ component("image_decoders") {
 
   if (enable_av1_decoder) {
     sources += [
-      "avif/crabbyavif_image_decoder.cc",
-      "avif/crabbyavif_image_decoder.h",
+      "avif/avif_image_decoder.cc",
+      "avif/avif_image_decoder.h",
     ]
-
-    deps += [ "//third_party/crabbyavif" ]
+    libs = [ "avif" ]
   }
 }
 
@@ -122,7 +121,7 @@ source_set("unit_tests") {
   }
 
   if (enable_av1_decoder) {
-    sources += [ "avif/crabbyavif_image_decoder_test.cc" ]
+    sources += [ "avif/avif_image_decoder_test.cc" ]
   }
 
   # TODO(crbug.com/40031409): Fix code that adds exit-time destructors and
--- a/third_party/blink/renderer/platform/image-decoders/image_decoder.cc
+++ b/third_party/blink/renderer/platform/image-decoders/image_decoder.cc
@@ -49,7 +49,7 @@
 #include "ui/gfx/geometry/size_conversions.h"
 
 #if BUILDFLAG(ENABLE_AV1_DECODER)
-#include "third_party/blink/renderer/platform/image-decoders/avif/crabbyavif_image_decoder.h"
+#include "third_party/blink/renderer/platform/image-decoders/avif/avif_image_decoder.h"
 #endif
 
 namespace blink {
@@ -205,7 +205,7 @@ String SniffMimeTypeInternal(scoped_refp
     return "image/bmp";
   }
 #if BUILDFLAG(ENABLE_AV1_DECODER)
-  if (CrabbyAVIFImageDecoder::MatchesAVIFSignature(fast_reader)) {
+  if (AVIFImageDecoder::MatchesAVIFSignature(fast_reader)) {
     return "image/avif";
   }
 #endif
@@ -315,7 +315,7 @@ std::unique_ptr<ImageDecoder> ImageDecod
                                                 max_decoded_bytes);
 #if BUILDFLAG(ENABLE_AV1_DECODER)
   } else if (mime_type == "image/avif") {
-    decoder = std::make_unique<CrabbyAVIFImageDecoder>(
+    decoder = std::make_unique<AVIFImageDecoder>(
         alpha_option, high_bit_depth_decoding_option, color_behavior, aux_image,
         max_decoded_bytes, animation_option);
 #endif
