diff -ruN vscode-9999.orig/build/gulpfile.vscode.js vscode-9999/build/gulpfile.vscode.js
--- a/build/gulpfile.vscode.js	2022-02-19 11:45:42.751433707 +0100
+++ b/build/gulpfile.vscode.js	2022-02-19 11:46:24.338088284 +0100
@@ -354,6 +354,7 @@
 	{ platform: 'linux', arch: 'x64' },
 	{ platform: 'linux', arch: 'armhf' },
 	{ platform: 'linux', arch: 'arm64' },
+	{ platform: 'linux', arch: 'ppc64' },
 ];
 BUILD_TARGETS.forEach(buildTarget => {
 	const dashed = (str) => (str ? `-${str}` : ``);
diff -ruN vscode-9999.orig/build/gulpfile.vscode.linux.js vscode-9999/build/gulpfile.vscode.linux.js
--- a/build/gulpfile.vscode.linux.js	2022-02-19 11:45:53.324763965 +0100
+++ b/build/gulpfile.vscode.linux.js	2022-02-19 11:48:09.361391093 +0100
@@ -23,7 +23,7 @@
 const linuxPackageRevision = Math.floor(new Date().getTime() / 1000);
 
 function getDebPackageArch(arch) {
-	return { x64: 'amd64', armhf: 'armhf', arm64: 'arm64' }[arch];
+	return { x64: 'amd64', armhf: 'armhf', arm64: 'arm64', ppc64: 'ppc64el' }[arch];
 }
 
 function prepareDebPackage(arch) {
@@ -117,7 +117,7 @@
 }
 
 function getRpmPackageArch(arch) {
-	return { x64: 'x86_64', armhf: 'armv7hl', arm64: 'aarch64' }[arch];
+	return { x64: 'x86_64', armhf: 'armv7hl', arm64: 'aarch64', ppc64: 'ppc64le' }[arch];
 }
 
 function prepareRpmPackage(arch) {
@@ -257,6 +257,7 @@
 	{ arch: 'x64' },
 	{ arch: 'armhf' },
 	{ arch: 'arm64' },
+	{ arch: 'ppc64' },
 ];
 
 BUILD_TARGETS.forEach(({ arch }) => {
